src/plugins/path/path.c

```diff
@@ -89,9 +269,14 @@ int elektraPathSet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key * par
 	int rc = 1;
 	while ((cur = ksNext (returned)) != 0)
 	{
-		const Key * meta = keyGetMeta (cur, "check/path");
-		if (!meta) continue;
+		const Key * pathMeta = keyGetMeta (cur, "check/path");
+		if (!pathMeta) continue;
```

Comment:
So if there is no check/path no permission check is done? This does not seem to be wanted?

Reply:
I assumed that the plugin enhancement is an extension ...
check/path -> check if the file exists at all
check/permission... -> check if the file has the correct permissions in addition

If the file is not existent, a permission check would not work.

I can "incorporate" the check/path functionality in beneath the check/permission/... part if this is desirable.

Reply:
If the key specifies a logfile for example the file would not need to exist, but the user has to have permission to create it. So permission checks without an existing file have applications, whether they are possible to implement I don't know. Possibly you would need to check permissions of the parent directory of the given path.

Reply:
If it is desirable that an application can write a log file into the directory, the user should check for the write permission on the directory, not on a non-existent file in my opinion

Reply:
But then you would need to separate keys, one for the path of the log directory (which is check with path and permission), and one for just the filename. That doesn't seem intuitive or user friendly to me.

I think the best solution is adding a `c` (create) permission. The `r`, `w`, `x` permissions automatically imply the normal `check/path` check. The `c` permission does imply the `check/path` check and `r` (not sure if needed) and `w` permissions for the parent directory of the given path. Only the combinations `c` and `cx` (also requires `x` permission on parent directory) would be allowed.

Reply:
This again turns out to be a little bit subjective. As an application developer, all I would want is that my application can write to a certain directory for log files (especially when thinking about the operations team where I work). Especially concerning rolling logs your approach might seem infeasible or cumbersome. 

Also I am not sure why I would need to have a separate key for the file. Can there be a case in which you create a file in a (for you) writable directory but immediately lose the right to write/read to the created file?

Apart from that it would be very troublesome and error-prone to implement ... if the parent directory does not exist yet, would I need to check for the parent directory of the directory recursively? Limiting the user to cx because he should assume that c = rw on the parent directory seems also a bit confusing to me.

We could keep your approach for the future if there is really a need for it. Maybe @markus2330  could help us on how to proceed.

Reply:
> Also I am not sure why I would need to have a separate key for the file. 

The second key is there, so that the filename can be changed.

> Especially concerning rolling logs your approach might seem infeasible or cumbersome.

Specifying just a directory is possible no matter, if we implement the `c` permission or not. 

But if an application only ever produces one logfile, it seems unnecessary to create a separate directory for each application, when all these directories will only contain a single file. I would much rather just use a single file under `/var/log`, which means 1) not using `check/permission` 2) specifying the filename separately 3) implementing `c` or something similar.

> if the parent directory does not exist yet
 
This would be an error. It is quite common to require that the directory in which a file shall be created must exist.

> Limiting the user to cx because he should assume that c = rw on the parent directory seems also a bit confusing to me.

Not sure I understood that correctly, but I just checked and `c` should actually `wx` on the parent directory (so `cx` doesn't make sense acutally). [Wikipedia](https://en.wikipedia.org/wiki/File_system_permissions#Traditional_Unix_permissions) states that `w` without `x` is meaningless on directories. And `r` is only needed to list filenames, not to read files with known names (`x` is needed for that).

> Apart from that it would be very troublesome and error-prone to implement ...

Before creating `modeMask` this should work:
```c
char * validPath = elektraStrDup(keyString (key));
if (strcmp(modes, "c") == 0)
{
    modes = "wx";
    char * lastSlash = strrchr(validPath, '/');
    *lastSlash = '\0';   
}
else if (strchr(modes, 'c') != NULL) 
{
    // TODO: error, c cannot be combined
}
```
You would also need to do in the appropriate places `elektraFree(validPath)`. If you want `c` and the original path in the error message instead of `wx` and the directory path, just use `keyString(userTypes)` and `keyString(key)` instead of `modes` and `validPath`.

Reply:
@kodebach thank you for the input!

My main problem was that the documentation of check/permission did not mention that it has anything to do with check/path. Also the name does not indicate such a relation (check/path/mode would be much better).

Having support for non-existing log files seem to be out-of-scope with the current progress. Better we have a more focused feature implemented correctly. (With good error messages.)